src/app/App.js

```diff
@@ -13,9 +14,9 @@ class App extends Component {
       <div className="App">
         <BrowserRouter>
           <Switch>
-            <Route exact path="/" component={Map} />
-            <Route path="/login" component={Login} />
-            <Route path="/form" component={Form} />
+            <Route exact path="/" component={withTracker(Map)} />
+            <Route path="/login" component={withTracker(Login)} />
+            <Route path="/form" component={withTracker(Form)} />
```

Comment:
Suggestion: This doesn't have to be an HOC and could exist as a standalone component which would avoid wrapping each component.

```jsx
import Tracker from "./Tracker";
// ...
<BrowserRouter>
   <Route path="/" component={Tracker} />
   <Switch>
       <Route exact path="/" component={Map} />
       <Route exact path="/" component={Login} />
       <Route exact path="/" component={Form} />
   </Switch>
</BrowserRouter>
```

Reply:
Yeah, I saw a bunch of discussions about this before going with the current solution.
It seems people use both approaches, but the HOC method is what appears in their official [demo](https://github.com/react-ga/react-ga/tree/master/demo) and [documentation](https://github.com/react-ga/react-ga/wiki/React-Router-v4-withTracker). Apparently, the reason it's preferred is that it gives you more flexibility in terms of tracking other components, beyond just the high-level routes. Having to repeat it in a few places didn't seem too bad to me, as long as it's mostly contained to this one file.

But, I don't feel very strongly about it either way. If you still think it's worth it, I can make the change.

Reply:
That's a good point. I don't mind either way. 👍